cmd/agent/app/snmp.go

```diff
@@ -293,7 +295,22 @@ func printValue(pdu gosnmp.SnmpPDU) error {
 	switch pdu.Type {
 	case gosnmp.OctetString:
 		b := pdu.Value.([]byte)
-		fmt.Printf("STRING: %s\n", string(b))
+		if !utilFunc.IsStringPrintable(b) {
+			var strValue string
+			strValue = fmt.Sprintf("%X", b)
+			if len(strValue)%2 != 0 {
+				strValue = "0" + strValue
+			}
+			var message string
+			message = "Hex-STRING: "
+			for i := 0; i <= len(strValue)-4; i += 2 {
+				message += strValue[i:i+2] + " "
+			}
+			message += strValue[len(strValue)-2:] + "\n"
+			fmt.Print(message)
```

Comment:
Not blocking, but maybe could be a bit simpler like this:

```suggestion
            var strBytes string
            for _, bt := b {
                strBytes = strBytes.append(hex.EncodeToString([]byte{bt}))
            }
            fmt.Print("Hex-STRING: " + strings.Join(strBytes, " "))
```